### PR TITLE
Introduce style_derive::cg

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18200)

### CODE REVIEW

**Review by emilio — APPROVED** [review]

File: components/style_derive/cg.rs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    prefix: &str,
+) -> (Tokens, Vec<BindingInfo<'a>>) {
+    synstructure::match_pattern(
+        &name,
+        &variant.data,
+        &BindOpts::with_prefix(BindStyle::Ref, prefix.to_owned()),
+    )
+}
+
+pub fn trait_parts<'a>(
```
Comment (emilio) [inline components/style_derive/cg.rs]: fwiw, it'd be nice to have docs in a bunch of these.

**Review by emilio — APPROVED** [review]

File: components/style_derive/cg.rs (hunk 1 above)
Comment (nox) [inline components/style_derive/cg.rs]: Will do when I'm done with the entire cleanup of that crate.

**Pull request merged**